events/tests/test_event_get.py

```diff
@@ -192,6 +192,47 @@ def test_get_event_list_verify_bbox_filter(api_client, event, event2):
     )
 
 
+@pytest.mark.django_db
+def test_get_event_list_verify_dwithin_filter(event, event2):
+    origin = Point(24, 24)
+    # Sanity check for distances
+    assert origin.distance(event.location.position) > 36
+    assert origin.distance(event2.location.position) < 34
+
+    # Don't filter if missing either origin or metres.
+    get_list_and_assert_events(f"dwithin_origin={origin.x},{origin.y}", [event, event2])
+    get_list_and_assert_events("dwithin_metres=35", [event, event2])
+
+    get_list_and_assert_events(
+        f"dwithin_origin={origin.x},{origin.y}&dwithin_metres=35", [event2]
+    )
+    get_list_and_assert_events(
+        f"dwithin_origin={origin.x},{origin.y}&dwithin_metres=37", [event, event2]
+    )
+
+    # Should work with a float distance as well.
+    get_list_and_assert_events(
+        f"dwithin_origin={origin.x},{origin.y}&dwithin_metres=35.01", [event2]
+    )
+
+
+@pytest.mark.parametrize(
+    "query_string",
+    [
+        "dwithin_origin=0,0&dwithin_metres=foo",
+        "dwithin_origin=0&dwithin_metres=35",
+        "dwithin_origin=0,0,0&dwithin_metres=35",
+        "dwithin_origin=0,foo&dwithin_metres=35",
+    ],
+)
+@pytest.mark.django_db
+def test_get_event_list_verify_dwithin_filter_error(
+    query_string, api_client, event, event2
+):
+    response = get_list_no_code_assert(api_client, query_string=query_string)
+    assert response.status_code == 400
```

Comment:
Because it is pretty easy to get a 400 response from DRF when something else is wrong, I would like that this also checked that we are actually hitting the error message we are excepting.